compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ConstantPropagation.java

```diff
@@ -440,6 +446,38 @@ public void visit(BLangCompoundAssignment compoundAssignNode) {
         result = compoundAssignNode;
     }
 
+    @Override
+    public void visit(BLangMatchStatement matchStatement) {
+        matchStatement.expr = rewrite(matchStatement.expr);
+        rewrite(matchStatement.matchClauses);
+        result = matchStatement;
+    }
+
+    @Override
+    public void visit(BLangMatchClause matchClause) {
+        for (BLangMatchPattern matchPattern : matchClause.matchPatterns) {
```

Comment:
can't we use the `rewrite` method used in line:452 here

Reply:
Changed